### PR TITLE
add a note wrt torch.nn.functional.scaled_dot_product_attention

### PR DESCRIPTION
**Pull request by lancerts**
followup change of https://github.com/pytorch/pytorch/pull/120565 

- Added a note in the transformer class pointing out the mask definition is opposite to that of :attr:`attn_mask` in
            torch.nn.functional.scaled_dot_product_attention.
@mikaylagawarecki

### CODE REVIEW

File: torch/nn/modules/transformer.py [hunk 1]
```diff
@@ -178,8 +178,10 @@ def forward(self, src: Tensor, tgt: Tensor, src_mask: Optional[Tensor] = None, t
 
             Note: [src/tgt/memory]_mask ensures that position :math:`i` is allowed to attend the unmasked
             positions. If a BoolTensor is provided, positions with ``True``
-            are not allowed to attend while ``False`` values will be unchanged. If a FloatTensor
-            is provided, it will be added to the attention weight.
+            are not allowed to attend while ``False`` values will be unchanged.
+            (This definition is opposite to that of :attr:`attn_mask` in
```
Comment (mikaylagawarecki) [inline torch/nn/modules/transformer.py:182]: Could you move this to a `..note::` above to make it more visible please! (maybe around line 63)

e.g. 
```
..note:: 

If a boolean tensor is provided for any of the mask arguments, positions with a ``True`` value are not allowed to participate in the attention, which is the opposite of the definition for :attr:`attn_mask` in :func:`torch.nn.functional.scaled_dot_product_attention`
```

Comment (lancerts) [reply torch/nn/modules/transformer.py]: Thanks for the suggestion. Moved the note right below the forward function since the mask args are for the forward function in the transformer.

File: torch/nn/modules/transformer.py [hunk 2]
```diff
@@ -132,6 +132,12 @@ def forward(self, src: Tensor, tgt: Tensor, src_mask: Optional[Tensor] = None, t
                 memory_is_causal: bool = False) -> Tensor:
         r"""Take in and process masked source/target sequences.
 
+        .. note::
+
+        If a boolean tensor is provided for any of the [src/tgt/memory]_mask arguments, positions with a ``True`` value are
+        not allowed to participate in the attention,
+        which is the opposite of the definition for :attr:`attn_mask` in :func:`torch.nn.functional.scaled_dot_product_attention`.
```
Comment (mikaylagawarecki) [inline torch/nn/modules/transformer.py:139]:
```suggestion
        .. note::

            If a boolean tensor is provided for any of the [src/tgt/memory]_mask arguments, positions with a ``True`` value are
            not allowed to participate in the attention,
            which is the opposite of the definition for :attr:`attn_mask` in :func:`torch.nn.functional.scaled_dot_product_attention`.
```

**Review by mikaylagawarecki — APPROVED** [review]
Thanks a lot!
